nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,83 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+
+router.get("/", async (req, res) => {
+  if(Object.keys(req.query).length === 0 ){
+  res.send(meals);
+} 
+else if
+(req.query.hasOwnProperty("id")){
+
+      const requestId =Number(req.query.id);
+       const idFinder=meals.some(meal=>meal.id==requestId)
+       const result=meals.filter(meal=>meal.id===requestId);
+    
+      if(idFinder){
+        res.send(result);
+      }else{
+        res.send(`There is no meal with id ${requestId}`);
+      }
+}
+else if
+(req.query.hasOwnProperty("maxPrice")){
+ 
+  const maxPrice =Number(req.query.maxPrice);
+  const maxPriceFinder=meals.some(meal=>meal.price <= maxPrice);
+  
+  if(maxPriceFinder){
+    res.send(meals.filter(meal=>meal.price <= maxPrice));
```

Comment:
A way to avoid searching 2 times for price <= maxPrice would be to only use filter and then if the returned array is empty you return your error message